doc/api/esm.md

```diff
@@ -1220,27 +1178,94 @@ loaded from disk but before Node.js executes it; and so on for any `.coffee`,
 `.litcoffee` or `.coffee.md` files referenced via `import` statements of any
 loaded file.
 
-## Resolution algorithm
+#### "import map" loader
+
+The previous two loaders defined `load` hooks. This is an example of a loader
+that does its work via the `resolve` hook. This loader reads an
+`import-map.json` file that specifies which specifiers to override to another
+URL (this is a very simplistic implemenation of a small subset of the
+"import maps" specification).
+
+```js
+// import-map-loader.js
+import fs from 'node:fs/promises';
+
+const { imports } = JSON.parse(await fs.readFile('import-map.json'));
+
+export async function resolve(specifier, context, nextResolve) {
+  if (Object.hasOwn(imports, specifier)) {
+    return nextResolve(imports[specifier], context);
+  }
+
+  return nextResolve(specifier, context);
+}
+```
+
+Let's assume we have these files:
+
+```js
+// main.js
+import 'a-module';
+```
+
+```json
+// import-map.json
+{
+  "imports": {
+    "a-module": "./some-module.js"
+  }
+}
+```
+
+```js
+// some-module.js
+console.log('some module!');
+```
+
+If you run `node --experimental-loader ./import-map-loader.js main.js`
+the output will be `some module!`.
+
+## Default resolution and loading algorithm
 
 ### Features
 
-The resolver has the following properties:
+The default resolver has the following properties:
 
 * FileURL-based resolution as is used by ES modules
-* Support for builtin module loading
 * Relative and absolute URL resolution
 * No default extensions
 * No folder mains
 * Bare specifier package resolution lookup through node\_modules
+* Does not fail on unknown extensions or protocols
+
+The default loader has the following properties
+
+* Support for builtin module loading via `node:` URLs
+* Support for "inline" module loading via `data:` URLs
+* Support for `file:` module loading
+* Fails on any other URL protocol
+* Fails on unknown extensions for `file:` loading
+  (supports only `.cjs`, `.js`, and `.mjs`)
 
-### Resolver algorithm
+### Default resolver algorithm
 
 The algorithm to load an ES module specifier is given through the
 **ESM\_RESOLVE** method below. It returns the resolved URL for a
 module specifier relative to a parentURL.
 
+The algorithm does not determine whether the resolved URL protocol can be
+loaded by Node.js, because other loaders may be able to load it.
+The default loader _does_ fail if it was asked to load a URL
+that has an unsuppported protocol (not `file:`, `data:`, or `node:`).
+
+The algorithm also tries to determine the format of the file based
+on the extension (see `ESM_FILE_FORMAT` algorithm below), but does
+not fail if it does not recognize the extension. The default loader
+_does_ fail if it was asked to load a file with an unsupported
+extension (not `.mjs`, `.cjs`, or `.json`).
```

Comment:
I'd prefer we are more explicit about the Node.js resolution algorithm concerns over loaders here which is what this specification is for.

```suggestion
The resolution algorithm determines the full resolved URL for a module
load, along with it's suggested module format. The resolution algorithm
does not determine whether the resolved URL protocol can be loaded,
or whether the file extensions are permitted, instead these validations are applied
by Node.js during the load phase (for example, if it was asked to load a URL
that has a protocol that is not `file:`, `data:`, or `node:`).

The algorithm also tries to determine the format of the file based
on the extension (see `ESM_FILE_FORMAT` algorithm below). If it does
not recognize the file extension (eg if it is not `.mjs`, `.cjs`, or `.json`), then
a format of `undefined` is returned, while will throw during the load phase.
```

Reply:
(Grammar): “its suggested module format.” (not “it’s”)

Also maybe: “a protocol that is not `file:`, `data:`, `node:`, or if `--experimental-network-imports` is enabled, `https:`.”